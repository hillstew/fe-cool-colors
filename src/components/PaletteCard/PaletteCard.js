import React from 'react';

export const PaletteCard = ({ palette, deletePalette, palettes }) => {
  const { name, color_1, color_2, color_3, color_4, color_5, id } = palette;
  return (
    <div className="PaletteCard">
      <p className="palette-name">{name}</p>
      <div className="palette-container">
        <div className="palette">
          <div style={{ backgroundColor: color_1 }} className="palette-box" />
          <div style={{ backgroundColor: color_2 }} className="palette-box" />
          <div style={{ backgroundColor: color_3 }} className="palette-box" />
          <div style={{ backgroundColor: color_4 }} className="palette-box" />
          <div style={{ backgroundColor: color_5 }} className="palette-box" />
        </div>
        <button
          className="delete-palette-button"
          onClick={() => deletePalette(id, palettes)}
        />
      </div>
    </div>
  );
};
