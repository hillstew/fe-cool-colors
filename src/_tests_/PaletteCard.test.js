import React from 'react';
import { PaletteCard } from '../components/PaletteCard/PaletteCard';
import { shallow } from 'enzyme';

const mockProps = {
  palette: {
      name: 'test-palette',
      color_1: '808080',
      color_2: '232323',
      color_3: '1a1a1a',
      color_4: 'aa2222',
      color_5: '123abc'
  },
  setColors: jest.fn(),
  deletePalette: jest.fn()
}

describe('PaletteCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<PaletteCard {...mockProps} />);
  });

  it('should pass the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
