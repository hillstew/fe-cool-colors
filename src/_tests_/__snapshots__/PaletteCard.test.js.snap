// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteCard should pass the snapshot 1`] = `
<div
  className="PaletteCard"
>
  <p>
    test-palette
  </p>
  <div
    className="palette"
  >
    <div
      className="palette-box"
      style={
        Object {
          "backgroundColor": "808080",
        }
      }
    />
    <div
      className="palette-box"
      style={
        Object {
          "backgroundColor": "232323",
        }
      }
    />
    <div
      className="palette-box"
      style={
        Object {
          "backgroundColor": "1a1a1a",
        }
      }
    />
    <div
      className="palette-box"
      style={
        Object {
          "backgroundColor": "aa2222",
        }
      }
    />
    <div
      className="palette-box"
      style={
        Object {
          "backgroundColor": "123abc",
        }
      }
    />
  </div>
  <button
    className="delete-palette-button"
  >
    Delete
  </button>
</div>
`;
